File: app/src/main/java/org/ngbp/jsonrpc4jtestharness/rpc/xLink/IXLink.java
package org.ngbp.jsonrpc4jtestharness.rpc.xLink;

import com.github.nmuzhichin.jsonrpc.annotation.JsonRpcMethod;
import com.github.nmuzhichin.jsonrpc.annotation.JsonRpcType;

import org.ngbp.jsonrpc4jtestharness.rpc.drm.model.NotifyParams;

@JsonRpcType
public interface IXLink {
    @JsonRpcMethod("org.atsc.notify")
    Object xLinkResolutionNotification();

    @JsonRpcMethod("org.atsc.xlinkResolution")
    NotifyParams xLinkResolved();
}

File: app/src/main/java/org/ngbp/jsonrpc4jtestharness/rpc/xLink/IXLink.kt
package org.ngbp.jsonrpc4jtestharness.rpc.xLink

import com.github.nmuzhichin.jsonrpc.annotation.JsonRpcMethod
import com.github.nmuzhichin.jsonrpc.annotation.JsonRpcType
import org.ngbp.jsonrpc4jtestharness.rpc.RpcResponse

@JsonRpcType
interface IXLink {
    @JsonRpcMethod("org.atsc.xlinkResolution")
    fun xLinkResolved(): RpcResponse
}